ui/app/js/controllers/rules_controller.js

```diff
@@ -103,7 +122,8 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
           $scope.pr_ch_options.sort().unshift("All rules");
           $scope.pr_ch_filter = "All rules";
           if ($scope.pr_ch_options.includes(localStorage.getItem('pr_ch_filter'))){
-            $scope.pr_ch_filter = localStorage.getItem('pr_ch_filter');
+            $scope.pr_ch_filter = $rootScope.pr_ch_filter || localStorage.getItem('pr_ch_filter') || "All rules";
+            $rootScope.pr_ch_filter = $scope.pr_ch_filter;
```

Comment:
Can you explain the rationale behind using $rootScope? I don't understand why it's needed. There's no reason to store both the explicit filter from a URL _and_ what's in localStorage. The logic should be:
- If there's a valid filter in the URL use that, and update localStorage with it.
- If there's a filter with an error in the URL, use "All rules", and update localStorage with it.
- If there's no filter in the URL, use what's in localStorage
- If there's no filter in the URL or in localStorage, use "All rules", and update localStorage with it.

And of course, whenever the user changes the filter, update both the URL and localStorage with it.

Please let me know if I'm missing something here.